lib/internal/streams/writable.js

```diff
@@ -629,31 +634,34 @@ Writable.prototype.end = function(chunk, encoding, cb) {
   }
 
   if (err) {
-    // Do nothing...
-  } else if (!state.errored && !state.ending) {
+    if (cb) {
```

Comment:
I think the performance regression is due to one of those `if (cb)` checks

Reply:
This is end function? Should not be in hot path?

Reply:
my bad